bindings/go/Makefile.am

```diff
@@ -0,0 +1,21 @@
+#
+# Copyright (C) Mellanox Technologies Ltd. 2021. ALL RIGHTS RESERVED.
+# See file LICENSE for terms.
+#
+
+if HAVE_GO
+
+GOPATH=$(abs_top_builddir)/bindings/go/build-go
+CGOCFLAGS="-I$(abs_top_builddir)/src -I$(top_srcdir)/src"
+CGOLDFLAGS="-L$(abs_top_builddir)/src/ucp/$(objdir) -lucp -L$(abs_top_builddir)/src/ucs/$(objdir) -lucs"
```

Comment:
In order to link wth full filename need to use `:filname`: 
> Note that this behavior does not apply to :filename, which always specifies a file called filename.
But [cgo doesn't support this characters](https://golang.org/cmd/cgo/):
> For security reasons, only a limited set of flags are allowed, notably -D, -U, -I, and -l. To allow additional flags, set CGO_CFLAGS_ALLOW to a regular expression matching the new flags. To disallow flags that would otherwise be allowed, set CGO_CFLAGS_DISALLOW to a regular expression matching arguments that must be disallowed. In both cases the regular expression must match a full argument: to allow -mfoo=bar, use CGO_CFLAGS_ALLOW='-mfoo.*', not just CGO_CFLAGS_ALLOW='-mfoo'. Similarly named variables control the allowed CPPFLAGS, CXXFLAGS, FFLAGS, and LDFLAGS.

> Also for security reasons, only a limited set of characters are permitted, notably alphanumeric characters and a few symbols, such as '.', that will not be interpreted in unexpected ways. Attempts to use forbidden characters will get a "malformed #cgo argument" error.

Also better to use `-L$(abs_top_builddir)/src/ucp/$(objdir) -lucp` if we want to build on non linux platforms someday, since shared lib would have different suffix there.